driver/src/main/java/org/neo4j/driver/v1/RecordView.java

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright (c) 2002-2015 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.neo4j.driver.v1;
+
+/**
+ * A record view provides access to an ordered, finite collection of zero or more fields.
+ *
+ * A field is a pair of a string key and a value.
+ *
+ * A record view must not contain multiple fields with the same key.
+ */
+public interface RecordView
```

Comment:
"done by methods on implementations" sounds very strange because it means we'll expose actual implementations. So we'll make it harder for ourselves to change the implementation in future.
If `Record` is immutable than it should only have methods for reading and if `RecordView` is mutable than it should inherit reading methods from `Record` and add writing methods.